force-app/src/classes/MethodSpy.cls

```diff
@@ -29,7 +29,7 @@ global class MethodSpy {
     this.callLog = new CallLog();
   }
 
-  public Object call(List<Object> args) {
+  public Object call(List<Type> paramTypes, List<String> paramNames, List<Object> args) {
```

Comment:
question/fix: Don't you think that we should keep `public` method operating and deprecated them instead of changing the signature? I'm concerned about the impact on users code when they will update those classes.

Reply:
I agree, nice catch !

Reply:
proposal: duplicate v1 test suite and keep it as a gold/master test suite to ensure that we have a retro-compatible API + deprecated old APIs or at least and a comment to make it clear.

Reply:
I'll propose something in this spirit soon